pep-9999.rst

```diff
@@ -0,0 +1,731 @@
+contain a string literal or a forward reference to some ``Color.RED``
+enum member?
+
+In cases like these, we always assume the user meant to construct a
+literal string. If the user wants a forward reference, they must wrap
+the entire literal type in a string -- e.g. ``"Literal[Color.RED]"``.
+
+Literals, enums, and Any
+------------------------
+
+Another ambiguity is when the user attempts to use some expression that
+is meant to be an enum but is actually of type ``Any``. For example,
+suppose a user attempts to import an enum from a package with no type hints::
+
+   from typing import Literal
+   from lib_with_no_types import SomeEnum  # SomeEnum has type 'Any'!
+
+   # x has type `Literal[Any]` due to the bad import
+   x: Literal[SomeEnum.FOO]
+
+Because ``Literal`` may not be parameterized by ``Any``, this program
+is *illegal*: the type checker should report an error with the last line.
+
+In short, while ``Any`` may effectively be used as a placeholder for any
+arbitrary *type*, it is currently **not** allowed to serve as a placeholder
+for any arbitrary *value*.
+
+
+Inferring types for literal expressions
+=======================================
+
+This section describes under what circumstances some expression should have
+an inferred Literal type.
+
+For example, under what circumstances should literal expressions like ``"blue"``
+have an inferred type of ``Literal["blue"]`` vs ``str``?
+
+In short, type checkers are expected to be conservative and bias towards
+inferring standard types like ``str``. Type checkers should infer ``Literal[...]``
+only in contexts where a Literal type is explicitly requested.
+
+Variable assignment
+-------------------
+
+When assigning a literal expression to an unannotated variable, the
+inferred type of the variable is the original base type, not ``Literal[...]``.
+For example::
+
+    border_color = "blue"
+    reveal_type(border_color)   # Revealed type is 'str'
+
+This helps ensure we don't break the semantics of any existing code.
+
+If the user wants ``foo`` to have a literal type, they must 
+explicitly add an annotation::
+
+   border_color: Literal["blue"] = "blue"
+   reveal_types(border_color)   # Revealed type is 'Literal["blue"]'
+
+Or alternatively, use the ``Final`` qualifier::
+
+   border_color: Final = "blue"
+   reveal_types(border_color)   # Revealed type is 'Final[Literal["blue"]]'
+
+The ``Final`` qualifier will automatically infer a ``Literal`` type in
+an assignment if the LHS is a literal expression, or an expression of
+type ``Literal[...]``.
+
+**TODO:** Link to the PEP draft for the ``Final`` qualifier once it's ready.
+
+**Note 1:** A potential third way of declaring a Literal might be to 
+try using ``Literal`` as a qualifier::
+
+   foo: Literal = "hello"   # Illegal!
+
+Type checkers should *reject* lines like these. Unlike ``Final`` and ``ClassVar``,
+``Literal`` is a *type*, not a *qualifier*. Only qualifiers should infer their
```

Comment:
While I think I intuitively grasp what "qualifier" is, in my opinion, it should be explained or defined. In mypy docs it's used only in the context of `Literal` and it's not defined there as well.